packages/editor/src/components/post-preview-button/index.js

```diff
@@ -138,59 +94,126 @@ export class PostPreviewButton extends Component {
+				width: '1200px',
+				height: '800px',
+			},
+		} );
+	}
+
+	setTabletPreview() {
+		this.setState( {
+			previewSize: {
+				width: '768px',
+				height: '1024px',
+			},
+		} );
+	}
+
+	setMobilePreview() {
+		this.setState( {
+			previewSize: {
+				width: '375px',
+				height: '812px',
+			},
+		} );
+	}
 
-		// Display a 'Generating preview' message in the Preview tab while we wait for the
-		// autosave to finish.
-		writeInterstitialMessage( this.previewWindow.document );
+	closePreviewModal() {
+		this.setState( { isPreviewOpen: false } );
 	}
 
 	render() {
 		const { previewLink, currentPostLink, isSaveable } = this.props;
-
 		// Link to the `?preview=true` URL if we have it, since this lets us see
 		// changes that were autosaved since the post was last published. Otherwise,
 		// just link to the post's URL.
 		const href = previewLink || currentPostLink;
 
 		return (
-			<Button
-				isLarge
-				className="editor-post-preview"
-				href={ href }
-				target={ this.getWindowTarget() }
-				disabled={ ! isSaveable }
-				onClick={ this.openPreviewWindow }
-			>
-				{ _x( 'Preview', 'imperative verb' ) }
-				<span className="screen-reader-text">
-					{
+			<>
+				<Button
+					isLarge
+					className="editor-post-preview"
+					disabled={ ! isSaveable }
+					onClick={ this.openPreviewModal }
+				>
+					{ _x( 'Preview', 'imperative verb' ) }
+
+					<span className="screen-reader-text">
+						{
 						/* translators: accessibility text */
-						__( '(opens in a new tab)' )
-					}
-				</span>
-			</Button>
+							__( '(opens in a new tab)' )
+						}
+					</span>
+				</Button>
+				{ this.state.isPreviewOpen &&
+					<Modal
+						title={
+							// translators: Preview dialog title.
+							__( 'Preview dialog' )
+						}
+						onRequestClose={ this.closePreviewModal }
+						// Needed so the Modal doesn't close when tabbing into the iframe.
+						shouldCloseOnClickOutside={ false }
+						className="editor-post-preview-button__preview-modal"
+					>
+						<div className="editor-post-preview__controls">
+							<IconButton
+								icon={ DesktopIcon }
+								onClick={ this.setDesktopPreview }
+								label={ __( 'Preview desktop screen' ) }
+							/>
+							<IconButton
+								icon={ TabletIcon }
+								onClick={ this.setTabletPreview }
+								label={ __( 'Preview tablet screen' ) }
+							/>
+							<IconButton
+								icon={ MobileIcon }
+								onClick={ this.setMobilePreview }
+								label={ __( 'Preview phone screen' ) }
+							/>
+							<Button
+								isLarge
+								className="editor-post-preview__new-tab"
+								href={ href }
+								target={ this.getWindowTarget() }
+								onClick={ this.openPreviewInNewTab }
+							>
+								{ __( 'Open preview in new tab' ) }
+							</Button>
+						</div>
+						<div className="editor-post-preview__frame-container">
+							<iframe
+								className="editor-post-preview__frame"
+								title={ __( 'Responsive preview frame' ) }
```

Comment:
Please remove the word "frame", that's already announced by assistive technologies because of the element native role